### DISCUSSION TITLE
launcher query 接口逻辑目前是只对 service_mgr 开放的吗？

### DISCUSSION THREAD

**Discussion by t0350** [cloudwu/skynet#1382] [Q&A] [answered]
看了下 launcher `QUERY` 接口直接用的 request 一端 session 来做查找指定服务的逻辑。
```
function command.QUERY(_, request_session)
	for address, session in pairs(launch_session) do
		if session == request_session then
			return address
		end
	end
end
```
这样很可能是会返回错误的服务的？可以简单地构造出 `launch_session` 中 `session` 相同的用例。

目前因为 `QUERY` 逻辑只在 service_mgr 服务调用，所以确实没有问题。
想问下是不是就是只考虑 launcher 该接口仅开放给 service_mgr 用，谢谢！

**Comment by cloudwu** [comment, marked as answer]
所有没有写在 wiki 上的东西，默认都是实现细节，不排除日后修改的可能。
